### PR TITLE
Fuzzy timer for exponential backoff and update polling

### PR DESCRIPTION
**Pull request by marineam**

### CODE REVIEW

File: omaha/client/fuzzytime.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+package client
+
+import (
+	"math/rand"
+	"time"
+)
+
+func init() {
+	// Ensure seeding the prng is never forgotten, that would defeat
```
Comment (yifan-gu) [inline omaha/client/fuzzytime.go]: s/prng/ping

Comment (marineam) [inline omaha/client/fuzzytime.go]: No, that's pseudo-rng.

Comment (yifan-gu) [inline omaha/client/fuzzytime.go]: sorry, you are right.

File: omaha/client/client.go [hunk 2]
```diff
@@ -93,6 +104,16 @@ func (c *Client) SetClientVersion(clientVersion string) {
 	c.clientVersion = clientVersion
 }
 
+// NextPing returns a timer channel that will fire when the next update
+// check or ping should be sent.
+func (c *Client) NextPing() <-chan time.Time {
```
Comment (yifan-gu) [inline omaha/client/client.go]: Didn't see where the function is used? It's in another PR?

Comment (marineam) [inline omaha/client/client.go]: It would be used by the code driving the client, I'll add an example.

Comment (yifan-gu) [inline omaha/client/client.go]: OK.

Comment (marineam) [inline omaha/client/client.go]: A basic example added in b447c8d

**Comment by yifan-gu** [comment]
LGTM

**Pull request merged**